docs/usage.rst

```diff
@@ -145,6 +149,90 @@ subset of the files within the that directory with the optional ``config['patch_
+package defines a python representation of this object, :py:class:`~fv3config.DiagTable`, which can
+be used to explicitly represent the ``diag_table`` within an fv3config configuration dictionary.
+
+The ``DiagTable`` object can be initialized from a dict (here serialized as YAML) as follows. Suppose
+the following is saved within ``sample_diag_table.yaml``:
+
+.. code-block:: yaml
+
+    name: example_diag_table
+    base_time: 2000-01-01 00:00:00
+    file_configs:
+    - name: physics_diagnostics
+      frequency: 1
+      frequency_units: hours
+      field_configs:
+      - field_name: totprcpb_ave
+        module_name: gfs_phys
+        output_name: surface_precipitation_rate
+      - field_name: ULWRFtoa
+        module_name: gfs_phys
+        output_name: upward_longwave_radiative_flux_at_toa
+
+Then a ``DiagTable`` object can be initialized as:
+
+.. code-block:: python
+
+    >>> import fv3config
+    >>> import yaml
+    >>> with open("sample_diag_table.yaml") as f:
+            diag_table_dict = yaml.safe_load(f)
+    >>> diag_table = fv3config.DiagTable.from_dict(diag_table_dict)
+    >>> print(diag_table)  # will output diag_table in format expected by Fortran model
+    example_diag_table
+    2000 1 1 0 0 0
+
+    "physics_diagnostics", 1, "hours", 1, "hours", "time"
+
+    "gfs_phys", "totprcpb_ave", "surface_precipitation_rate", "physics_diagnostics", "all", "none", "none", 2
+    "gfs_phys", "ULWRFtoa", "upward_longwave_radiative_flux_at_toa", "physics_diagnostics", "all", "none", "none", 2
+
+The same ``DiagTable`` can also be initialized programmatically as follows:
+
+.. code-block:: python
+
+    >>> import fv3config
+    >>> import datetime
+    >>> diag_table = fv3config.DiagTable(
+            name="example_diag_table",
+            base_time=datetime.datetime(2000, 1, 1)
+            file_configs=[
+                fv3config.DiagFileConfig(
+                    name="physics_diagnostics",
+                    frequency=1,
+                    frequency_units="hours",
+                    field_configs=[
+                        fv3config.DiagFieldConfig(
+                            "gfs_phys",
+                            "totprcb_ave"
+                            "surface_precipitation_rate"
+                        ),
+                        fv3config.DiagFieldConfig(
+                            "gfs_phys",
+                            "ULWRFtoa"
+                            "upward_longwave_radiative_flux_at_toa"
+                        ),
+                    ]
+                )
+            ]
+        )
+
+String representations of the ``diag_table`` (i.e. those expected by the Fortran model) can be parsed
+with the :py:meth:`fv3config.DiagTable.from_str` method.
+
+If serializing an ``fv3config`` configuration object to yaml, it is recommended to use the
+:py:meth:`fv3config.DiagTable.asdict` method before dumping the configuration dictionary. This
```

Comment:
Why is this recommended, and what are the dangers if you do not do this? What exactly does "use the asdict method before dumping" mean? Do you mean to replace the diagtable with a dict representation using asdict?

You may want to recommend foremost that the reader uses `config_to_yaml`, and that if they need to serialize manually then they can use asdict to get the primitive representation.

No change required for merge but if you think you can quickly make it clearer please do.

Reply:
Sounds good, focused paragraph on `config_to_yaml`. Might be worth adding a separate section on serializing/deserializing fv3config config dicts later on.